engine/class_modules/sc_shaman.cpp

```diff
@@ -8405,6 +8493,25 @@ void shaman_t::trigger_lightning_shield( const action_state_t* state )
   }
 }
 
+void shaman_t::trigger_lava_surge(const player_t* source ) {
+  if ( buff.lava_surge->check() )
+  {
+    proc.wasted_lava_surge->occur();
+  }
+
+  proc.lava_surge->occur();
+  if ( !source->executing || source->executing->id != 51505 )
```

Comment:
Replace with if ( !executing || !executing->id != 51505 )

Reply:
Done. See next commit. Thank you